ra/ra.go

```diff
@@ -2679,9 +2679,13 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 // createPendingAuthz checks that a name is allowed for issuance and creates the
 // necessary challenges for it and puts this and all of the relevant information
 // into a corepb.Authorization for transmission to the SA to be stored
-func (ra *RegistrationAuthorityImpl) createPendingAuthz(reg int64, identifier identifier.ACMEIdentifier) (*corepb.Authorization, error) {
-	authz := &corepb.Authorization{
-		Identifier:     identifier.Value,
+func (ra *RegistrationAuthorityImpl) createPendingAuthz(reg int64, identifier identifier.ACMEIdentifier) (*sapb.NewAuthzRequest, error) {
+	authz := &sapb.NewAuthzRequest{
+		IdentifierValue: identifier.Value,
+		Identifier: &sapb.Identifier{
+			Type:  string(identifier.Type),
+			Value: identifier.Value,
+		},
 		RegistrationID: reg,
 		Status:         string(core.StatusPending),
 		Expires:        timestamppb.New(ra.clk.Now().Add(ra.pendingAuthorizationLifetime).Truncate(time.Second)),
```

Comment:
I'm not terribly fond of this pattern where we construct an object and assign some of its fields later depending on various conditions. There's no reason to make a change to this now, it was like this already, but it stood out to me as a potential source of problems in the future.

Reply:
Completely agreed. In the follow-up to this PR, where I clean up all of the old fields, I've simplified this entire function to
```go
	challTypes, err := ra.PA.ChallengeTypesFor(identifier)
	if err != nil {
		return nil, err
	}

	challStrs := make([]string, len(challTypes))
	for i, t := range challTypes {
		challStrs[i] = string(t)
	}

	authz := &sapb.NewAuthzRequest{
		Identifier: &sapb.Identifier{
			Type:  string(identifier.Type),
			Value: identifier.Value,
		},
		RegistrationID: reg,
		Expires:        timestamppb.New(ra.clk.Now().Add(ra.pendingAuthorizationLifetime).Truncate(time.Second)),
		ChallengeTypes: challStrs,
		Token:          core.NewToken(),
	}

	return authz, nil
```

Does leaving this cleanup for the follow-up work for you?